umap/static/umap/js/modules/saving.js

```diff
@@ -0,0 +1,54 @@
+export class SaveManager {
+  constructor() {
+    this._queue = new Set()
+  }
+
+  get isDirty() {
+    return Boolean(this._queue.size)
+  }
+
+  async save() {
+    for (const obj of this._queue) {
+      const ok = await obj.save()
+      if (!ok) break
+      this.delete(obj)
+    }
+  }
+
+  add(obj) {
+    this._queue.add(obj)
+    this.checkStatus()
+  }
+
+  delete(obj) {
+    this._queue.delete(obj)
+    this.checkStatus()
+  }
+
+  has(obj) {
+    return this._queue.has(obj)
+  }
+
+  checkStatus() {
+    document.body.classList.toggle('umap-is-dirty', this._queue.size)
+  }
+}
+
+export const SAVEMANAGER = new SaveManager()
+
+export class ServerStored {
```

Comment:
I'm not sure of that pattern, a proxy to a constant makes me think we are missing a JS way of handling it 😅

Reply:
Are you sure we are not using PHP ? :thinking: 

I'll look for a better pattern, but if you have a suggestion, please go… :)

Reply:
Note that I had a look at this bible verset, and stopped reading here https://stackoverflow.com/a/59646297 ;)

Reply:
Also, one of my goals was to move that responsability out of the main `Map` object, to make it lighter and easier to refactor, but this ends with the needs to another object, so maybe you'd prefer keeping it in the `Map` object (once separated from the `LeafletMap`) ?